base/logging_context.go

```diff
@@ -74,10 +78,69 @@ func bucketCtx(parent context.Context, b Bucket) context.Context {
 
 // bucketNameCtx extends the parent context with a bucket name.
 func bucketNameCtx(parent context.Context, bucketName string) context.Context {
-	parentLogCtx, _ := parent.Value(LogContextKey{}).(LogContext)
+	parentLogCtx, _ := parent.Value(logContextKey).(LogContext)
 	newCtx := LogContext{
 		TestName:       parentLogCtx.TestName,
 		TestBucketName: bucketName,
 	}
-	return context.WithValue(parent, LogContextKey{}, newCtx)
+	return LogContextWith(parent, &newCtx)
+}
+
+// ContextAdderKey is the type used to store custom data in the go context
+type ContextAdderKey int
+
+const (
+	logContextKey ContextAdderKey = iota
```

Comment:
Can we come up with a more descriptive name for logContextKey (and LogContext), now that they aren't the only context?  

In fact, I wonder if LogContextKey is a better name to use for ContextAdderKey.